### PR TITLE
Fixed notifications in android Oreo

### PR DESCRIPTION
**Pull request by feelfreelinux**
Some users reported problems with notifications not showing. It happens because we use android sdk 26, which requires notification channels.
- Moved dagger injector to compation object
- Created new `NotificationHelper`

### CODE REVIEW

File: versions.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ext {
     versions = [
             buildTools      : '26.0.2',
-            androidPlugin   : '3.0.0-beta7',
+            androidPlugin   : '3.0.0-rc1',
```
Comment (rowysock) [inline versions.gradle]: Does it really belong to this issue?

File: app/src/main/kotlin/com/wabadaba/dziennik/MainApplication.kt [hunk 2]
```diff
@@ -10,8 +10,9 @@ import com.wabadaba.dziennik.di.MainComponent
 import javax.inject.Inject
 
 open class MainApplication : MultiDexApplication() {
-
-    lateinit var mainComponent: MainComponent
```
Comment (rowysock) [inline app/src/main/kotlin/com/wabadaba/dziennik/MainApplication.kt]: Please make changes like these in separate PRs.

File: app/src/main/kotlin/com/wabadaba/dziennik/api/notification/NotificationHelper.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package com.wabadaba.dziennik.api.notification
+
+import android.app.Notification
+import android.app.NotificationManager
+import android.app.NotificationChannel
+import android.os.Build
+import android.support.annotation.RequiresApi
+
+
+class NotificationHelper(private val notificationManager: NotificationManager) {
```
Comment (rowysock) [inline app/src/main/kotlin/com/wabadaba/dziennik/api/notification/NotificationHelper.kt]: Helper suffix doesn't convey any meaning. I see no point in having this class. Please move it's contents to LibrusGcmListenerService

File: app/src/main/kotlin/com/wabadaba/dziennik/di/ApplicationModule.kt [hunk 4]
```diff
@@ -67,4 +70,10 @@ class ApplicationModule(private val mainApplication: Application) {
     @Provides
     @Singleton
     fun provideGPServicesChecker(): GPServicesChecker = GPServicesChecker()
+
+    @Provides
+    fun provideNotificationManager(context: Context) = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
```
Comment (rowysock) [inline app/src/main/kotlin/com/wabadaba/dziennik/di/ApplicationModule.kt]: I see no point in having NotificationManager provided by Dagger. Unless it helps with testing, please put this inside LibrusGcmListenerService

File: app/src/main/kotlin/com/wabadaba/dziennik/api/notification/NotificationHelper.kt [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    companion object {
+        val NOTIFICATION_CHANNEL_ID = "com.wabadaba.dziennik.notifications"
+        val NOTIFICATION_TAG = "com.wabadaba.dziennik"
```
Comment (rowysock) [inline app/src/main/kotlin/com/wabadaba/dziennik/api/notification/NotificationHelper.kt]: shouldn't this be read from some kind of android service?

Comment (feelfreelinux) [reply app/src/main/kotlin/com/wabadaba/dziennik/api/notification/NotificationHelper.kt]: These are notification channels id / tag. You need to set them manually

File: app/src/main/kotlin/com/wabadaba/dziennik/api/notification/NotificationHelper.kt (hunk 3 above)
Comment (feelfreelinux) [reply app/src/main/kotlin/com/wabadaba/dziennik/api/notification/NotificationHelper.kt]: Changed suffix. Having this code in LibrusGcmListenerService would violate the [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single_responsibility_principle)

Comment (feelfreelinux) [reply app/src/main/kotlin/com/wabadaba/dziennik/api/notification/NotificationHelper.kt]: And this class allows us to simply reuse this notification channel.
